implementation/src/main/java/io/smallrye/mutiny/operators/UniCache.java

```diff
@@ -70,17 +73,42 @@ private synchronized void onCancellation(UniSubscriber<? super I> subscriber) {
         subscribers.remove(subscriber);
     }
 
-    private void replay(UniSubscriber<? super I> subscriber) {
-        synchronized (this) {
-            if (state != COMPLETED) {
```

Comment:
This check failed in tests since it races when subscriptions are done in parallel of which one triggers a resubscription but the others would get a replay.
That's why I merged the former replay into the `COMPLETED` action itself so there's no need for this check in my opinion.